src/services/v2/on-chain/mover/debit-card/DebitCardOnChainService.ts

```
@@ -76,7 +130,83 @@ export class DebitCardOnChainService extends MoverOnChainService {
     }
 
-    return undefined;
+    return inputAsset;
+  }
+
+  public async getUnwrappedTokenPrice(
+    inputAsset: SmallToken,
+    inputAssetPrice: string
+  ): Promise<string> {
+    const specialTokenHandler = this.specialTokenHandlers.find((h) =>
+      h.canHandle(inputAsset.address, this.network)
+    );
+
+    if (specialTokenHandler !== undefined) {
+      const unwrappedTokenPrice = await this.getPriceByAddress(
+        specialTokenHandler.getUnwrappedToken().address
+      );
+      return unwrappedTokenPrice;
+    }
+
+    return inputAssetPrice;
+  }
+
+  public async getUnwrappedData(
+    inputAsset: SmallTokenInfo,
+    inputAmount: string,
+    inputAssetPrice: string
+  ): Promise<UnwrappedData> {
+    const specialTokenHandler = this.specialTokenHandlers.find((h) =>
+      h.canHandle(inputAsset.address, this.network)
+    );
+
+    if (specialTokenHandler === undefined) {
+      return {
+        unwrappedToken: inputAsset,
+        amountInWei: toWei(inputAmount, inputAsset.decimals),
+        unwrappedTokenPrice: inputAssetPrice
+      };
+    }
+
+    const newAmount = await specialTokenHandler.getUnwrappedAmount(inputAmount);
+    const unwrappedToken = specialTokenHandler.getUnwrappedToken();
+
+    const unwrappedTokenPrice = await this.getPriceByAddress(
+      unwrappedToken.address
+    );
+
+    return {
+      unwrappedToken: unwrappedToken,
+      amountInWei: getInteger(toWei(newAmount, unwrappedToken.decimals)),
+      unwrappedTokenPrice: unwrappedTokenPrice
+    };
+  }
+
+  public async getWrappedDataByUnwrapped(
+    wrappedAsset: SmallTokenInfo,
+    unwrappedAsset: SmallTokenInfo,
+    unwrappedAmount: string
+  ): Promise<WrappedData> {
+    const specialTokenHandler = this.specialTokenHandlers.find((h) =>
+      h.canHandle(wrappedAsset.address, this.network)
+    );
+
+    if (specialTokenHandler === undefined) {
+      // unwrappedAsset should be equal to wrappedAsset
```

Let's raise a Sentry exception here with a detailed breadcrumbs the moments before
Our just make an assertion that will throw an error if it fails